### PR TITLE
Polymer 2.x

### PR DESCRIPTION
**Pull request by bergie**
We went to Polymer 1.x recently (#668). This pushes further, to a Polymer release generation that is properly available on NPM.

See:

* [Polymer 2.x upgrade guide](https://www.polymer-project.org/2.0/docs/upgrade)
* [noflo-polymer CHANGES](https://github.com/noflo/noflo-polymer#changes)

Tasks:

* [x] Update to new WebComponents polyfill
* [x] Use the new `<slot>` API
* [x] Solve issue with Font Awesome icons (CSS doesn't apply inside shadow DOM)
* [x] Fix styling

### CODE REVIEW

**Comment by bergie** [comment]
screenshot 2017-11-10 at 21 01 39

Styling generally looks fine now, the main remaining issue is how to load styles for CodeMirror and the-graph, since we [can't load regular CSS files](https://www.polymer-project.org/2.0/docs/devguide/style-shadow-dom#external-stylesheets) inside Shadow DOM.

**Comment by bergie** [comment]
As a bonus item, passing styling information to elements via CSS variables means now theming works better throughout

screenshot 2017-11-10 at 22 57 46

**Comment by bergie** [comment]
Also just did a quick testing round on iOS. Seems to work fine.

File: elements/noflo-graph-inspector.html [hunk 1]
```diff
@@ -272,18 +284,18 @@ <h1 on-keydown="checkUpdateName" on-blur="updateName" contenteditable="">{{graph
           return;
         }
         this.htmlEditor = CodeMirror(this.$.html_editor, {
-          lineNumbers: true,
+          lineNumbers: false,
```
Comment (jonnor) [inline elements/noflo-graph-inspector.html]: @bergie why lline numbers are disabled? This seems unrelated

File: elements/noflo-graph-inspector.html [hunk 2]
```diff
@@ -272,18 +284,18 @@ <h1 on-keydown="checkUpdateName" on-blur="updateName" contenteditable="">{{graph
...
+          theme: 'mdn-like',
         });
       },
       prepareTestsEditor: function () {
         if (!this.spec) {
           return;
         }
         this.testsEditor = CodeMirror(this.$.tests_editor, {
-          lineNumbers: true,
+          lineNumbers: false,
```
Comment (jonnor) [inline elements/noflo-graph-inspector.html]: Same, why disabled?

File: elements/the-graph-nav.html [hunk 3]
```diff
@@ -94,52 +94,14 @@
...
           this.set('style.height', this.height + 'px');
-          // HACK way to make PolymerGestures work for now
-          if (window.PolymerGestures) {
-            var noop = function () {
-            };
-            window.PolymerGestures.addEventListener(this, 'track', noop);
-            window.PolymerGestures.addEventListener(this, 'trackend', noop);
-            window.PolymerGestures.addEventListener(this, 'tap', noop);
-          }
-          // Pan graph by dragging map
```
Comment (jonnor) [inline elements/the-graph-nav.html]: @bergie is this feature removed or what?

File: elements/the-graph-nav.html [hunk 4]
```diff
@@ -94,52 +94,14 @@
...
-            this.fire('panto', panTo);
-            event.preventTap();
-          }.bind(this));
-          this.addEventListener('trackend', function (event) {
-            // Don't pan graph
-            event.stopPropagation();
-          }.bind(this));
-          // Tap to fit
-          this.addEventListener('tap', function () {
-            this.fire('triggerfit');
```
Comment (jonnor) [inline elements/the-graph-nav.html]: And this feature?

File: elements/noflo-graph-inspector.html (hunk 1 above)
Comment (bergie) [reply elements/noflo-graph-inspector.html]: It seemed to trigger a rendering bug with CodeMirror when used in a narrow space

With line numbering:

screenshot 2017-11-13 at 21 11 30

Without:

screenshot 2017-11-13 at 21 11 41

File: elements/noflo-graph-inspector.html (hunk 2 above)
Comment (bergie) [reply elements/noflo-graph-inspector.html]: See [above](#discussion_r150651831)

File: elements/the-graph-nav.html (hunk 3 above)
Comment (bergie) [reply elements/the-graph-nav.html]: It seems this never worked after we went Polymer 1.x

Comment (jonnor) [reply elements/the-graph-nav.html]: Might not have anything to do with Polymer 1.x, but with the-graph. There is a plain-React way to do it alo, see the-graph full example.

File: elements/noflo-graph-inspector.html (hunk 1 above)
Comment (jonnor) [reply elements/noflo-graph-inspector.html]: Editing component code without line numbers is a bit shit...

Comment (bergie) [reply elements/noflo-graph-inspector.html]: This is not for component code, only for graph preview HTML (and graph specs)
